modules/table/src/main/java/org/apache/ignite/internal/table/distributed/StorageUpdateHandler.java

```diff
@@ -252,10 +254,34 @@ private void tryRemoveFromIndexes(BinaryRow rowToRemove, RowId rowId, Cursor<Rea
         }
     }
 
-    private void removeFromIndex(BinaryRow row, RowId rowId) {
-        for (TableSchemaAwareIndexStorage index : indexes.get().values()) {
-            index.remove(row, rowId);
-        }
+    /**
+     * Tries removing partition's oldest stale entry and its indexes.
+     *
+     * @param lowWatermark Low watermark for the vacuum.
+     */
+    public boolean vacuum(HybridTimestamp lowWatermark) {
+        return storage.runConsistently(() -> {
+            BinaryRowAndRowId vacuumed = storage.pollForVacuum(lowWatermark);
+
+            if (vacuumed == null) {
+                // Nothing was garbage collected.
+                return false;
+            }
+
+            BinaryRow binaryRow = vacuumed.binaryRow();
+
+            assert binaryRow != null;
```

Comment:
Looks like there might be a mistake and we can remove the tombstone.

Reply:
Tombstones can't be returned from pollForVacuum. They're removed along with a non-tombstone value

Reply:
It's interesting that if you add a test to `org.apache.ignite.internal.storage.AbstractMvPartitionStorageGcTest`, then everyone except `RocksDbMvPartitionStorageGcTest` will pass it, in general this is a possible situation.    
```
    @Test
    void testTombstone() {
        addAndCommit(null);

        addAndCommit(TABLE_ROW);

        addAndCommit(TABLE_ROW2);

        BinaryRowAndRowId row = pollForVacuum(HybridTimestamp.MAX_VALUE);

        assertNull(row.binaryRow());
    }
```

Reply:
That's not relevant to this ticket, but I think there's no harm in fixing it

Reply:
I do not agree with your reasoning, technically the situation above is possible, it must be corrected at the beginning.

Reply:
I've already fixed it :)

Reply:
Thanks!